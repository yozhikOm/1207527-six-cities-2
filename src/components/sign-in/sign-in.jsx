import React from 'react';
import PropTypes from 'prop-types';
import {Header} from '../header/header.jsx';

const SignIn = (props) => {
  let emailInput = ``;
  let loginInput = ``;
  const {authenticateUser, history} = props;

  const isEmptyField = (field) => {
    if (field === undefined ||
      field === null) {
      console.warn(`Field is undefined or null`);
      return true;
    }
    if (field.value === null ||
      field.value.trim() === ``) {
      return true;
    }
    return false;
  };

  const redirect = () =>{
    const {history} = props;
    history.push(`/`);
  }
  
  const onSubmitHandler = (evt) => {
    if (evt) {
      evt.preventDefault();
    }
    if (!isEmptyField(emailInput) && !isEmptyField(loginInput)) {
      authenticateUser(emailInput.value, loginInput.value, redirect);
    } else {
      alert(`Вы забыли ввести логин и/или пароль. Пожалуйста, попробуйте еще раз`);
    }
  };
  return (
    <div className="page page--gray page--login">
      <Header isAuthorizationRequired={true} userInfo={null} />
      <main className="page__main page__main--login">
        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>
            <form className="login__form form" action="#" method="post" onSubmit={onSubmitHandler}>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">E-mail</label>
                <input ref={(element) => {
                  emailInput = element;
                }} className="login__input form__input" type="email" name="email" placeholder="Email" required="" />
              </div>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">Password</label>
                <input ref={(element) => {
                  loginInput = element;
                }} className="login__input form__input" type="password" name="password" placeholder="Password" required="" />
              </div>
              <button className="login__submit form__submit button" type="submit">Sign in</button>
            </form>
          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <a className="locations__item-link" href="#">
                <span>Amsterdam</span>
              </a>
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

SignIn.propTypes = {
  authenticateUser: PropTypes.func,
};

export {SignIn};
