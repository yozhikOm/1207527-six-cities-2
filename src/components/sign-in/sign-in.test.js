import React from 'react';
import {SignIn} from './sign-in.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';

it(`SignIn компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();

  renderer.render(<SignIn
    isAuthorizationRequired={true}
    authenticateUser={jest.fn()}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
