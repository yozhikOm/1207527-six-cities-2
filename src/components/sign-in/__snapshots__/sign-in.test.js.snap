// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn компонент рендерится корректно 1`] = `
<div
  className="page page--gray page--login"
>
  <Header
    isAuthorizationRequired={true}
    userInfo={null}
  />
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          action="#"
          className="login__form form"
          method="post"
          onSubmit={[Function]}
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              name="email"
              placeholder="Email"
              required=""
              type="email"
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              name="password"
              placeholder="Password"
              required=""
              type="password"
            />
          </div>
          <button
            className="login__submit form__submit button"
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </section>
    </div>
  </main>
</div>
`;
